hudi-common/src/main/java/org/apache/hudi/common/util/queue/BoundedInMemoryExecutor.java

```diff
@@ -54,29 +56,35 @@
   private final List<BoundedInMemoryQueueProducer<I>> producers;
   // Consumer
   private final Option<BoundedInMemoryQueueConsumer<O, E>> consumer;
+  // pre-execute function to implement environment specific behavior before executors (producers/consumer) run
+  private final Runnable preExecute;
```

Comment:
nit: may be we can name this preExecuteRunnable.

Reply:
fixed